### PR TITLE
EOS- 27644 Test and qualify granular deployment - Data Pod(Don't Merge)

### PR DESCRIPTION
**Pull request by zahidsk**

### CODE REVIEW

File: k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/cluster-template.yaml [hunk 1]
```diff
@@ -11,32 +11,6 @@ cluster:
       - name: hare
     storage:
     <<.Values.cluster.storage_list>>
-  - name: server_node
```
Comment (ujjwalpl) [inline k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/cluster-template.yaml:14]: You dont have to modify existing template and delete things from there. Full deployment will not work. 
Create new templates for data only

File: k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/config-template.yaml [hunk 2]
```diff
@@ -40,34 +40,8 @@ cortx:
       device_certificate: /etc/cortx/solution/ssl/stx.pem
   utils:
     message_bus_backend: kafka
-  s3:
```
Comment (ujjwalpl) [inline k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/config-template.yaml:43]: You dont have to modify existing template and delete things from there. Full deployment will not work.
Create new templates for data only

File: k8_cortx_cloud/deploy-cortx-cloud.sh [hunk 3]
```diff
@@ -643,46 +643,46 @@ function deployCortxConfigMap()
...
-        ./parse_scripts/subst.sh $cluster_server_node_file "cortx.node.name" "cortx-server-headless-svc-${node_name_list[$i]}"
-        uuid_str=$(UUID=$(uuidgen); echo ${UUID//-/})
-        ./parse_scripts/subst.sh $cluster_server_node_file "cortx.pod.uuid" "$uuid_str"
-        ./parse_scripts/subst.sh $cluster_server_node_file "cortx.svc.name" "cortx-server-headless-svc-${node_name_list[$i]}"
-        ./parse_scripts/subst.sh $cluster_server_node_file "cortx.node.type" "server_node"
-        # Create data machine id file for cortx server
-        auto_gen_node_path="$cfgmap_path/auto-gen-${node_name_list[$i]}-$namespace/server"
-        mkdir -p $auto_gen_node_path
-        echo $uuid_str > $auto_gen_node_path/id
+        # # Generate cluster server node file with type server_node in "node-info" folder
```
Comment (ujjwalpl) [inline k8_cortx_cloud/deploy-cortx-cloud.sh:646]: Why is this code commented?

File: k8_cortx_cloud/deploy-cortx-cloud.sh [hunk 4]
```diff
@@ -1313,10 +1313,10 @@ deployCortxLocalBlockStorage
 deleteStaleAutoGenFolders
 deployCortxConfigMap
 deployCortxSecrets
-deployCortxControl
+# deployCortxControl
```
Comment (ujjwalpl) [inline k8_cortx_cloud/deploy-cortx-cloud.sh:1316]: Code should support both full or data only deployment. Not just data only.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

**Comment by zahidsk** [comment]
Raising new PR with new template files created for Data pod

**Pull request closed without merging**